app/views/questionnaire.py

```diff
@@ -224,6 +236,44 @@ def get_thank_you(eq_id, form_type):  # pylint: disable=unused-argument
     return _redirect_to_latest_location(routing_path, collection_id, metadata.get('eq_id'), metadata.get('form_type'))
 
 
+@post_submission_blueprint.route('view-submission', methods=['GET'])
+@login_required
+def get_view_submission(eq_id, form_type):  # pylint: disable=unused-argument
+
+    session_data = get_session_store().session_data
+    g.schema = load_schema_from_params(session_data.eq_id, session_data.form_type)
+
+    if _is_submission_viewable(g.schema.json, session_data.submitted_time):
+        submitted_data = current_app.eq['submitted_responses'].get_item(session_data.tx_id)
```

Comment:
we need to check here is `submitted_responses` in present
`if 'submitted_responses' in current_app.eq:`

Reply:
Added check to `def is_view_submitted_response_enabled(schema):` so that we don't attempt anything around submitted answers unless avaliable.